tests/lib/rules/no-eval.js

```diff
@@ -42,6 +42,8 @@ ruleTester.run("no-eval", rule, {
         { code: "function foo() { var eval = 'foo'; globalThis[eval]('foo') }", env: { es2020: true } },
         "this.noeval('foo');",
         "function foo() { 'use strict'; this.eval('foo'); }",
+        { code: "'use strict'; this.eval('foo');", parserOptions: { ecmaFeatures: { globalReturn: true } } },
+        { code: "this.eval('foo');", parserOptions: { ecmaVersion: 6, sourceType: "module" } },
```

Comment:
> At the top level of scripts, this always refers to the global object, regardless of "use strict".

shouldn't it be an error? it's in esm, not script, so `this` is `undef`, `this.eval` will cause a runtime error: "Cannot read properties of undefined".

Reply:
I see. it's not an `no-eval` error, but we could add the (invalid) case in `no-invalid-this`?

Reply:
Yes, this shouldn't be a `no-eval` error because it isn't really an eval call.

I've added the test case to `no-invalid-this` tests in https://github.com/eslint/eslint/pull/15712/commits/bb0f0938279e4b4d4fa10d5388ef4d91ad203881